### DISCUSSION TITLE
Is it possible to use div instead of p?

### DISCUSSION THREAD

**Discussion by roeycohen** [facebook/lexical#3756] [Q&A]
I've tried creating a DivNode that inherits from ParagraphNode, but the default node seems to be P

Thanks for this great project :)

**Comment by trueadm** [comment]
You'll need to use the node replacement API to set the ParagraphNode to be your DivNode. https://lexical.dev/docs/concepts/node-replacement

**Reply by roeycohen** [reply]
hi @trueadm, thanks for your reply.

I've ended up overriding Paragraph's importDOM and converting existing div element to p.

**Reply by trueadm** [reply]
That sounds a bit hacky. Did the node replacement not work for you?

**Reply by roeycohen** [reply]
I didn't try... I found myself overriding all of these functions anyway...

TextNode.importDOM (for colors)
ParagraphNode.importDOM (div => p)
ListItemNode.importDOM (text-align support)

ParagraphNode.prototype.exportDOM (for better directionality saving)
TextNode.prototype.exportDOM (replacing spans with b, u, i and s to reduce html size)
ListItemNode.prototype.exportDOM (directionality + value remove)
LinkNode.prototype.exportDOM (prevent auto-links from being saved as regular links)

We were using draftjs till now, and were delighted to found out that lexical is already quite evolved and takes the same approach that we loved in draft :)

**Reply by edishu** [reply]
@trueadm the example you have shared adds styling to `CustomParagraphNode` node. But how can I convert it to a `div` element?

**Reply by etrepum** [reply]
You can change the tag used by overriding the createDOM, updateDOM, importDOM, and exportDOM methods of your ParagraphNode subclass

**Reply by edishu** [reply]
Thanks, that works well! However, I just `createDOM` only. Do I need to add  `updateDOM`, `importDOM`, and `exportDOM` too?

**Reply by etrepum** [reply]
You'll need importDOM if you ever want HTML import of your customized tag to create paragraph nodes, which may be relevant for copy and paste in certain situations. The other ones you can probably rely on the default implementations.
